RFS/src/test/java/com/rfs/cms/WorkCoordinatorTest.java

```diff
@@ -0,0 +1,179 @@
+//                });
+//        var doc2 = client.execute(workCoordinator.makeGetRequest(docId), r -> {
+//            return objMapper.readTree(r.getEntity().getContent());
+//        });
+//        Assertions.assertEquals("noop", response2.get("result").textValue(),
+//                "response that came back was unexpected - document == " + objMapper.writeValueAsString(doc2));
+//        Assertions.assertEquals(1, doc2.get("_source").get("numAttempts").longValue());
+//
+//        Thread.sleep(2500);
+//
+//        var response3 = client.execute(
+//                workCoordinator.makeUpdateRequest(docId, "node_1", Instant.now(), 2),
+//                r -> {
+//                    Assertions.assertEquals(HttpStatus.SC_OK, r.getCode());
+//                    return objMapper.readTree(r.getEntity().getContent());
+//                });
+//        Assertions.assertEquals("updated", response3.get("result").textValue());
+//        var doc3 = client.execute(workCoordinator.makeGetRequest(docId), r -> {
+//            return objMapper.readTree(r.getEntity().getContent());
+//        });
+//        Assertions.assertEquals(2, doc3.get("_source").get("numAttempts").longValue());
+//        Assertions.assertTrue(
+//                doc2.get("_source").get("expiration").longValue() <
+//                        doc3.get("_source").get("expiration").longValue());
+//
+//        var response4 = client.execute(
+//                workCoordinator.makeCompletionRequest(docId, "node_1", Instant.now()), r -> {
+//                    Assertions.assertEquals(HttpStatus.SC_OK, r.getCode());
+//                    return objMapper.readTree(r.getEntity().getContent());
+//                });
+//        var doc4 = client.execute(workCoordinator.makeGetRequest(docId), r -> {
+//            return objMapper.readTree(r.getEntity().getContent());
+//        });
+//        Assertions.assertEquals("updated", response4.get("result").textValue());
+//        Assertions.assertTrue(doc4.get("_source").get("completedAt").longValue() > 0);
+//        log.info("doc4="+doc4);
+    }
+
+    //@Test
+    public void testAcquireLeaseForQuery() throws Exception {
+        var objMapper = new ObjectMapper();
+        final var NUM_DOCS = 40;
+        try (var workCoordinator = new OpenSearchWorkCoordinator(httpClientSupplier.get(),
+                3600, "docCreatorWorker")) {
+            for (var i = 0; i < NUM_DOCS; ++i) {
+                final var docId = "R" + i;
+                workCoordinator.createUnassignedWorkItem(docId);
+            }
+        }
+
+        for (int runs=0; runs<2; ++runs) {
+            final var seenWorkerItems = new ConcurrentHashMap<String, String>();
+            var allFutures = new ArrayList<CompletableFuture<Void>>();
+            final var expiration = Duration.ofSeconds(5);
+            for (int i = 0; i < NUM_DOCS; ++i) {
+                int finalI = i;
+                int finalRuns = runs;
```

Comment:
Not actually final ...?

Reply:
whoops - that was an intellij auto-fix for putting a mutating value into a lambda.  The latest code cons-es an effectively final string outside of the lambda to workaround the underlying issue.